src/canvas-view/canvas/canvas-directive.js

```diff
@@ -0,0 +1,440 @@
+(function () {
+  'use strict';
+
+  angular.module('patternfly.canvas')
+  .filter('trustAsResourceUrl', ['$sce', function ($sce) {
+    return function (val) {
+      return $sce.trustAsResourceUrl(val);
+    };
+  }])
+
+  //
+  // Directive that generates the rendered chart from the data model.
+  //
+  .directive('pfCanvas', function ($document) {
+    return {
```

Comment:
Again, I agree, but it will take a significant rewrite of that core directive in order to convert it to a component.  The main pfCanvasEditor is a component and is what end users can/will access from a hybrid angular 2 setup.    @dgutride and I talked about that child/sub directives used in component templates were ok (not ideal), as long as the parent was a component.